File: ChimpokomonPython/chimpokomon/GimnasioChimpokomon.py
from ChimpokomonPython.chimpokomon.Zapato import Zapato
import ChimpokomonPython.logger.LoggerAux as LoggerAux
import ChimpokomonPython.ataque.Ataque as Ataque

class GimnasioChimpokomon():

    def __init__(self, chimpokomonA, chimpokomonB):
        self.chimpokomonA = chimpokomonA
        self.chimpokomonB = chimpokomonB
        
    log1 = LoggerAux()

    def peleaChimpokomones(self, chimpokomonA, chimpokomonB):
        while not chimpokomonA.estaDerrotado() and not chimpokomonB.estaDerrotado():
            chimpokomonA.pelearContra(chimpokomonB)
            chimpokomonB.pelearContra(chimpokomonA)
        if chimpokomonB.estaDerrotado():
             LoggerAux.info("El chimpokomon " + chimpokomonA.getNombre() + " ha ganado")
        else:
            LoggerAux.info("El chimpokomon " + chimpokomonB.getNombre() + " ha ganado")

    def main(self):
        ataque1 = [Ataque()]
        ataque2 = [Ataque()]

        ataque1.insert(Zapatazo())
        ataque2.insert(Zapatazo())
        ataque2.insert(PomadaWassington())

        gimnasio = GimnasioChimpokomon(Zapato(3, ataque1), Zapato(1, ataque2))

        gimnasio.peleaChimpokomones()

        print(gimnasio.log1.debug())

File: ChimpokomonPython/chimpokomon/GimnasioChimpokomon.java
package chimpokomon;

import Ataque.Ataque;
import Ataque.Zapatazo;
import Ataque.PomadaWassington;
import logger.*;

import java.util.ArrayList;
import java.util.List;

public class GimnasioChimpokomon {

    protected Chimpokomon chinpokomonA;
    protected Chimpokomon chinpokomonB;
    Logger logger = Logger.getInstance(new Debug());

    public GimnasioChimpokomon(Chimpokomon chinpokomonA, Chimpokomon chinpokomonB) {
        this.chinpokomonA = chinpokomonA;
        this.chinpokomonB = chinpokomonB;

    }

    public void peleaChinpokomonDe_Con_(){
        while(!this.chinpokomonA.estaDerrotado() && !this.chinpokomonB.estaDerrotado()){
            chinpokomonA.pelearContra(chinpokomonB);
            chinpokomonB.pelearContra(chinpokomonA);
        }
        if(chinpokomonB.estaDerrotado()){
            //System.out.printf("El ganador es "+ chinpokomonA.getNombre(),"\n");
            LoggerAux.info("El ganador es: "+ chinpokomonA.getNombre());
            LoggerAux.error("Este es un mensaje de error");
            logger.info("El ganador es: "+ chinpokomonA.getNombre());

        }
        else{
            //System.out.printf("El ganador es "+ chinpokomonB.getNombre(),"\n");
            //LoggerAux.info("El ganador es: "+ chinpokomonB.getNombre());
            logger.info("El ganador es: "+ chinpokomonB.getNombre());
            //LoggerAux.error("Este es un mensaje de error");
            //System.out.printf(logger.getInfo());

        }
    }

    public static void main(String[] args) {

        List<Ataque> ataque1 = new ArrayList<Ataque>();
        List<Ataque> ataque2 = new ArrayList<Ataque>();

        ataque1.add(new Zapatazo());
        ataque2.add(new Zapatazo());
        ataque2.add(new PomadaWassington());

        GimnasioChimpokomon gimnasioChimpokomon = new GimnasioChimpokomon(new Zapato(ataque2), new Zapato(ataque1));

        gimnasioChimpokomon.peleaChinpokomonDe_Con_();

        //gimnasioChimpokomon.logger.setLevel(new Warn());

        Logger logger = Logger.getInstance(new Debug());

        logger.setLevel(new Warn());

        System.out.println(logger.getLevel());





    }
}